src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -841,14 +842,24 @@ public override void Initialize(AnalysisContext context)
         public void TestReportingUnsupportedDiagnostic()
         {
             string source = @"";
-            var analyzers = new DiagnosticAnalyzer[] { new AnalyzerReportingUnsupportedDiagnostic() };
+            CSharpCompilation compilation = CreateCompilationWithMscorlib45(source);
+
+            var analyzer = new AnalyzerReportingUnsupportedDiagnostic();
+            var analyzers = new DiagnosticAnalyzer[] { analyzer };
             string message = new ArgumentException(string.Format(CodeAnalysisResources.UnsupportedDiagnosticReported, AnalyzerReportingUnsupportedDiagnostic.UnsupportedDescriptor.Id), "diagnostic").Message;
+            IFormattable context = $@"Exception occurred with following context:
+Compilation: {compilation.AssemblyName}
 
-            CreateCompilationWithMscorlib45(source)
+{new LazyToString(() => analyzer.ThrownException)}
+-----
+
+Suppress the following diagnostics to disable this analyzer: ID_1";
```

Comment:
This is nice. Do we want to force the unit test to run only on English locale with a conditional fact attribute?

Reply:
Updated to support other languages